main.cpp

```diff
@@ -18,13 +18,23 @@ using namespace clang::ast_matchers;
 using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
+    Rewriter& _rewriter;
+
 public:
-    CastCallBack(Rewriter& rewriter) {
+    CastCallBack(Rewriter& rewriter) : _rewriter(rewriter) {
         // Your code goes here
     };
 
-    void run(const MatchFinder::MatchResult &Result) override {
+    void run(const MatchFinder::MatchResult& Result) override {
         // Your code goes here
+        if (auto castExpr = Result.Nodes.getNodeAs<clang::CStyleCastExpr>("cast")) {
+            auto lParen = castExpr->getLParenLoc();
+            auto rParen = castExpr->getRParenLoc();
+
+            _rewriter.ReplaceText(lParen, 1, "static_cast<");
+            _rewriter.ReplaceText(rParen, 1, ">(");
+            _rewriter.InsertTextAfterToken(castExpr->getEndLoc(), ")");

```

Comment:
Please describe how to change the program code to get the following result:
```cpp
int main() {
    float fooooo;
    int i = static_cast<int>(fooooo)      ;
    return 0;
}
```
 for the origin `test.cpp`:
```cpp
int main() {

    float fooooo;

    int i = (int)          fooooo      ;

    return 0;

}
```

Reply:
Remove the opening bracket from _rewriter.ReplaceText(rParen, 1, ">(");
To add it, getSubExprAsWritten()->getBeginLoc() is usually used to get the initial position of the variable, then "(" is inserted at this token.